src/m365/spo/commands/tenant/tenant-applicationcustomizer-get.spec.ts

```diff
@@ -0,0 +1,364 @@
+      options: {
+        title: title,
+        id: id,
+        clientSideComponentId: clientSideComponentId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when no options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when title and id options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        title: title,
+        id: id
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when title and clientSideComponentId options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        title: title,
+        clientSideComponentId: clientSideComponentId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when id and clientSideComponentId options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        id: id,
+        clientSideComponentId: clientSideComponentId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if id is a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: id } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passed validation when title specified', async () => {
+    const actual = await command.validate({ options: { title: title } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if clientSideComponentId is valid', async () => {
+    const actual = await command.validate({ options: { clientSideComponentId: clientSideComponentId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('throws error when tenant app catalog doesn\'t exist', async () => {
+    const errorMessage = 'No app catalog URL found';
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${spoUrl}/_api/SP_TenantSettings_Current`) {
+        return { CorporateCatalogUrl: null };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, {
+      options: {
+        title: title
+      }
+    }), new CommandError(errorMessage));
+  });
+
+  it('retrieves an application customizer by title', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${spoUrl}/_api/SP_TenantSettings_Current`) {
+        return { CorporateCatalogUrl: appCatalogUrl };
+      }
+
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, listUrl);
+      if (opts.url === `${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/items?$filter=Title eq '${title}'`) {
```

Comment:
Rather than copying the same code that we've got in the command, let's specify the actual literal string so that we can verify our assumptions how the URL should be encoded. Let's do the same in other places where we repeat this pattern.